### PR TITLE
Transparent unavailable state in history

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

- [x] Use transparent color for unavailable state to make distinction between inactive and unavailable
- [x] Only use color when the state is active in state-badge in order to default to dark blue color when the entity is inactive.

**Example**
CleanShot 2022-12-12 at 12 35 05
 
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by spacegaier** [comment]
Will transparent not look like a rendering bug?

**Comment by piitaya** [comment]
We already have empty timeline when the entity doesn't exist and we the line is not display when we have unavailable state for numeric sensor for example.

Does it look like a rendering bug for you?

**Comment by matthiasdebaat** [comment]
I like it! How does it look like with the pop up?

**Comment by piitaya** [comment]
CleanShot 2022-12-12 at 13 17 21

File: src/panels/lovelace/cards/hui-entity-card.ts [hunk 1]
```diff
@@ -133,8 +133,9 @@ export class HuiEntityCard extends LitElement implements LovelaceCard {
 
     const name = this._config.name || computeStateName(stateObj);
 
-    const colored = stateObj && this.getStateColor(stateObj, this._config);
-    const active = stateObj && colored && stateActive(stateObj);
+    const active = stateObj && stateActive(stateObj);
+    const colored =
+      stateObj && active && this.getStateColor(stateObj, this._config);
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-entity-card.ts:138]:
```suggestion
    const colored = active && this.getStateColor(stateObj, this._config);
```

File: src/panels/lovelace/cards/hui-button-card.ts [hunk 2]
```diff
@@ -156,8 +156,9 @@ export class HuiButtonCard extends LitElement implements LovelaceCard {
       ? this._config.name || (stateObj ? computeStateName(stateObj) : "")
       : "";
 
-    const colored = stateObj && this.getStateColor(stateObj, this._config);
-    const active = stateObj && colored && stateActive(stateObj);
+    const active = stateObj && stateActive(stateObj);
+    const colored =
+      stateObj && active && this.getStateColor(stateObj, this._config);
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-button-card.ts:161]:
```suggestion
    const colored = active && this.getStateColor(stateObj, this._config);
```

File: src/components/entity/state-badge.ts [hunk 3]
```diff
@@ -107,27 +107,25 @@ export class StateBadge extends LitElement {
...
+        }
+        if (stateObj.attributes.brightness) {
+          const brightness = stateObj.attributes.brightness;
+          if (typeof brightness !== "number") {
+            const errorMessage = `Type error: state-badge expected number, but type of ${
+              stateObj.entity_id
+            }.attributes.brightness is ${typeof brightness} (${brightness})`;
+            // eslint-disable-next-line
+            console.warn(errorMessage);
           }
```
Comment (bramkragten) [inline src/components/entity/state-badge.ts:126]: Not sure we should even warn about this but ok 🤷‍♂️

File: src/components/chart/timeline-chart/textbar-element.ts [hunk 4]
```diff
@@ -37,8 +37,11 @@ export class TextBarElement extends BarElement {
     }
     const textColor =
       options.textColor ||
-      (options.backgroundColor &&
-        (luminosity(hex2rgb(options.backgroundColor)) > 0.5 ? "#000" : "#fff"));
+      (options.backgroundColor && options.backgroundColor === "transparent"
```
Comment (bramkragten) [inline src/components/chart/timeline-chart/textbar-element.ts:40]:
```suggestion
      (options?.backgroundColor === "transparent"
```

**Review by bramkragten — APPROVED** [review]

**Comment by spacegaier** [comment]
> We already have empty timeline when the entity doesn't exist and we the line is not display when we have unavailable state for a numeric sensor for example.
> 
> Does it look like a rendering bug for you?

I do think this looks like a bug, because there is no indication of what is going on. The user would need to have the knowledge that this could either mean "unavailable" or "entity not found" (instead of rendering bug), but I do think we should make that clear to the user rather than have them guess.

image

**Pull request merged**